plugins/woocommerce/includes/wc-conditional-functions.php

```diff
@@ -95,7 +95,7 @@ function is_product() {
 	function is_cart() {
 		$page_id = wc_get_page_id( 'cart' );
 
-		return ( $page_id && is_page( $page_id ) ) || Constants::is_defined( 'WOOCOMMERCE_CART' ) || wc_post_content_has_shortcode( 'woocommerce_cart' );
+		return ( $page_id && is_page( $page_id ) ) || Constants::is_defined( 'WOOCOMMERCE_CART' ) || wc_post_content_has_shortcode( 'woocommerce_cart' ) || apply_filters( 'woocommerce_is_cart', false );
```

Comment:
Building on my earlier note about documenting this new hook...

We're following an existing pattern in the sense we are copying the way this works inside the `is_checkout` function. However, it's arguably an unusual way of doing things and something like this may be better (or at least closer to the way these sorts of filter are usually applied):

```php
// Make an initial determination.
$is_cart = ( $page_id && is_page( $page_id ) ) || Constants::is_defined( 'WOOCOMMERCE_CART' ) || wc_post_content_has_shortcode( 'woocommerce_cart' );

// Then allow filter functions to override if necessary.
return apply_filters( 'woocommerce_is_cart', $is_cart );
```

In this case, given the precedent of the existing `woocommerce_is_checkout` hook, I think it makes sense to do things just as you have proposed—but it would definitely be worthwhile making this clear in the docblock for the hook.

(In other words, we should make it crystal clear that callbacks can only override a negative determination, and not a positive one.)